deeplearning4j/deeplearning4j-nn/src/main/java/org/deeplearning4j/util/ValidationUtils.java

```diff
@@ -0,0 +1,318 @@
+package org.deeplearning4j.util;
+
+import java.util.Arrays;
+import org.nd4j.base.Preconditions;
+
+public class ValidationUtils {
+
+    private ValidationUtils(){
+
+    }
+
+    /**
+     * Checks that the values is >= 0.
+     *
+     * @param data An int
+     * @param paramName The param name, for error reporting
+     */
+    public static void validateNonNegative(int data, String paramName){
+        Preconditions.checkArgument(data >= 0,
+                "Values for %s must be >= 0, got: %s", paramName, data);
+    }
+
+    /**
+     * Checks that the values is >= 0.
+     *
+     * @param data An int
+     * @param paramName The param name, for error reporting
+     */
+    public static void validateNonNegative(double data, String paramName){
+        Preconditions.checkArgument(data >= 0,
+                "Values for %s must be >= 0, got: %s", paramName, data);
+    }
+
+    /**
+     * Checks that all values are >= 0.
+     *
+     * @param data An array
+     * @param paramName The param name, for error reporting
+     */
+    public static void validateNonNegative(int[] data, String paramName){
```

Comment:
Handling of nulls? You have some in the later methods, but not here...
If null is acceptable, we should return null here. If not, let's either add a lombok ```@NonNull``` or a ```Preconditions.checkNotNull(```. At present, nulls will give a non-useful NPE.